core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -493,35 +495,43 @@ public void addSplits(String tableName, SortedSet<Text> splits)
       Map<KeyExtent,List<Text>> tabletSplits =
           mapSplitsToTablets(tableName, tableId, tabLocator, splitsTodo);
 
-      ArrayList<Pair<TFateId,List<Text>>> opids = new ArrayList<>(tabletSplits.size());
+      List<CompletableFuture<Pair<TFateId,List<Text>>>> futures = new ArrayList<>();
 
       final ByteBuffer EMPTY = ByteBuffer.allocate(0);
 
+      ExecutorService executor =
+          context.threadPools().getPoolBuilder("addSplits").numCoreThreads(16).build();
+
       // begin the fate operation for each tablet without waiting for the operation to complete
       for (Entry<KeyExtent,List<Text>> splitsForTablet : tabletSplits.entrySet()) {
-        var extent = splitsForTablet.getKey();
+        CompletableFuture<Pair<TFateId,List<Text>>> future = CompletableFuture.supplyAsync(() -> {
+          var extent = splitsForTablet.getKey();
 
-        List<ByteBuffer> args = new ArrayList<>();
-        args.add(ByteBuffer.wrap(extent.tableId().canonical().getBytes(UTF_8)));
-        args.add(extent.endRow() == null ? EMPTY : TextUtil.getByteBuffer(extent.endRow()));
-        args.add(extent.prevEndRow() == null ? EMPTY : TextUtil.getByteBuffer(extent.prevEndRow()));
-        splitsForTablet.getValue().forEach(split -> args.add(TextUtil.getByteBuffer(split)));
+          List<ByteBuffer> args = new ArrayList<>();
+          args.add(ByteBuffer.wrap(extent.tableId().canonical().getBytes(UTF_8)));
+          args.add(extent.endRow() == null ? EMPTY : TextUtil.getByteBuffer(extent.endRow()));
+          args.add(
+              extent.prevEndRow() == null ? EMPTY : TextUtil.getByteBuffer(extent.prevEndRow()));
+          splitsForTablet.getValue().forEach(split -> args.add(TextUtil.getByteBuffer(split)));
 
-        try {
-          handleFateOperation(() -> {
-            TFateInstanceType t = FateInstanceType.fromNamespaceOrTableName(tableName).toThrift();
-            TFateId opid = beginFateOperation(t);
-            executeFateOperation(opid, FateOperation.TABLE_SPLIT, args, Map.of(), false);
-            opids.add(new Pair<>(opid, splitsForTablet.getValue()));
-            return null;
-          }, tableName);
-        } catch (TableExistsException | NamespaceExistsException | NamespaceNotFoundException e) {
-          throw new RuntimeException(e);
-        }
+          try {
+            return handleFateOperation(() -> {
+              TFateInstanceType t = FateInstanceType.fromNamespaceOrTableName(tableName).toThrift();
+              TFateId opid = beginFateOperation(t);
+              executeFateOperation(opid, FateOperation.TABLE_SPLIT, args, Map.of(), false);
+              return new Pair<>(opid, splitsForTablet.getValue());
+            }, tableName);
+          } catch (TableExistsException | NamespaceExistsException | NamespaceNotFoundException
+              | AccumuloSecurityException | TableNotFoundException | AccumuloException e) {
+            throw new RuntimeException(e);
+          }
+        }, executor);
```

Comment:
Not sure how much time is spent serially making the second set of RPCs.  Those could also be thrown in a thread pool.  Since the code is already using completeable future this may be a small change.  May be able to do something like the following where `waitExecutor` is the 2nd thread pool. This would immediately queue the wait and finish RPCs to run in a 2nd thread pool once the RPCs to seed are done. 

```suggestion
        }, executor).thenAcceptAsync(entry->{
          final TFateId opid = entry.getFirst();
          final List<Text> completedSplits = entry.getSecond();
          // TODO wait for Fate op and then finish it
        }, waitExecutor);;
```

Reply:
Good idea. I was looking into using `thenAcceptAsync` for the wait but dismissed it since it would cause contention with the first operation. I think this approach with a second thread pool is a good idea. I'll try it out.